### PR TITLE
Use testdata everywhere

### PR DESCRIPTION
**Pull request by zbraniecki**
Here's a switch to use testdata for everything.

I'm starting it as draft to think about if I want to add several locales to the selection

### CODE REVIEW

File: components/datetime/tests/fixtures/tests/styles.json [hunk 1]
```diff
@@ -101,7 +101,7 @@
             }
         },
         "output": {
-            "value": "sobota, 21 marca 2020 08:25:07 zzzz"
+	    "value": "субота, 21 сакавіка 2020 'г'. 'у' 08:25:07, zzzz"
```
Comment (sffc) [inline components/datetime/tests/fixtures/tests/styles.json:104]: Looks like a small bug; I think you are supposed to remove the quotes.  Can you file a follow-up issue?

Comment (sffc) [inline components/datetime/tests/fixtures/tests/styles.json:104]: Please fix indentation.  I guess `cargo fmt` doesn't touch json files?

File: components/fs-data-provider/tests/test_file_io.rs [hunk 2]
```diff
@@ -5,15 +5,15 @@ use std::borrow::Cow;
...
-    let provider = FsDataProvider::try_new("tests/testdata/json_plurals_37")
+    let provider = FsDataProvider::try_new("../../resources/testdata/data/json")
         .expect("Loading file from testdata directory");
     let response = provider
         .load(&DataRequest {
             data_key: icu_data_key!(plurals: cardinal@1),
             data_entry: DataEntry {
                 variant: None,
                 // TODO: Migrate to LanguageIdentifier macro
-                langid: "sr".parse().expect("Valid language tag"),
```
Comment (sffc) [inline components/fs-data-provider/tests/test_file_io.rs:16]: We might as well stick with `"sr"` since we have it now.

**Review by sffc — APPROVED** [review]
Approved with some suggestions

**Review by sffc — APPROVED** [review]

File: components/datetime/tests/fixtures/tests/styles.json (hunk 1 above)
Comment (zbraniecki) [reply components/datetime/tests/fixtures/tests/styles.json]: https://github.com/unicode-org/icu4x/issues/319

**Pull request merged**